Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerToContextConnection.h

```diff
@@ -59,10 +55,11 @@ class NetworkConnectionToWebProcess;
 class WebSharedWorker;
 class WebSharedWorkerServer;
 
-class WebSharedWorkerServerToContextConnection final : public IPC::MessageSender, public IPC::MessageReceiver {
+class WebSharedWorkerServerToContextConnection final : public IPC::MessageSender, public IPC::MessageReceiver, public RefCounted<WebSharedWorkerServerToContextConnection> {
```

Comment:
I don't think this PR is safe. When making an object RefCounted, you need to look at data members to make sure extending the lifetime of the object is safe. `WeakRef<NetworkConnectionToWebProcess>` in particular is likely not OK and likely needs to become a WeakPtr with null checks.

Reply:
oh good point.